lib/streamer/cbor_streamer.go

```diff
@@ -0,0 +1,199 @@
+package streamer
+
+import (
+	"io"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/spacemonkeygo/errors"
+	"github.com/ugorji/go/codec"
+)
+
+var _ Mux = &CborMux{}
+
+type CborMux struct {
+	// the fd tip handles append; reads all use ReadAt and track their offsets individually
+	file  *os.File
+	codec *codec.Encoder
+	wmu   sync.Mutex
+}
+
+type cborMuxRow struct {
+	Label int    `json:"l"`
+	Msg   []byte `json:"m,omitempty"`
+	Sig   int    `json:"x,omitempty"` // 1->closed
+}
+
+func CborFileMux(filePath string) Mux {
+	file, err := os.OpenFile(filePath, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0755)
+	if err != nil {
+		panic(errors.IOError.Wrap(err))
+	}
+	file.Write([]byte{codec.CborStreamArray})
```

Comment:
Yep. An indefinite length array, to be precise.

Currently it's never closed.  One of the rough edges that should maybe be fixed.  Design blocker for doing so sanely: would have to change the mux interface to have a known-ahead-of-time set of labels.  Right now, all the readers just leave when they hit the close signal for their labels, so nobody's ever around to see the stream (not) end correctly.